content/python/upgrade_virtual_guest_dedicated.py.md

```diff
@@ -0,0 +1,421 @@
+---
+title: "Working with upgrade virtual guest dedicated"
+description: "A few examples on interacting with upgrade virtual guest."
+date: "2019-03-27"
+classes:
+    - "SoftLayer_Virtual_Guest"
+    - "SoftLayer_Product_Order"
+    - "SoftLayer_Container_Product_Order_Virtual_Guest_Upgrade"
+tags:
+    - "virtualserver"
+---
+
+Upgrade a VSI(cpu, ram)
+```
+import json
+
+import SoftLayer
+
+client = SoftLayer.create_client_from_env()
+orderService = client['SoftLayer_Product_Order']
+
+# The id of the virtual guest you wish upgrade
+virtualGuestId = 11111
+
+# Package keyName.
+package_keyname = 'CLOUD_SERVER'
+
+object_mask = 'mask[prices]'
+
+# items KeyNames to be upgrade
+items = ['GUEST_PRIVATE_CORES_8', 'RAM_12_GB']
+
+response_package = client["SoftLayer_Product_Package"].getAllObjects()
+
+package_id = None
+for val in response_package:
+    if val.get('keyName') == package_keyname:
+        package_id = val.get('id')
+
+get_item_id_price = client["SoftLayer_Product_Package"].getItems(mask=object_mask, id=package_id)
```

Comment:
wouldn't it better to use getUpgradeItemPrices? 
Not all items can be used to upgrade and I think we should show the users what method returns the valid items

Reply:
It is valid to use the method "getItems" or "getItemPrices" too, to upgrade a vs, it is for that I used that method.
The method "getUpgradeItemPrices" is another option for the customer that can be use.